homeassistant/components/landisgyr_heat_meter/__init__.py

```diff
@@ -53,3 +53,38 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    # Removing domain name and config entry id from entity unique id's, replacing it with device number
+    if config_entry.version == 1:
+
+        config_entry.version = 2
+
+        device_number = config_entry.data["device_number"]
+
+        # add label to prevent other devices with same number to conflict
+        new_unique_id = f"heat_meter_{device_number}"
```

Comment:
What other devices may have the same number?

Reply:
Yes, the unique id only needs to be unique within the integration.

The config entry unique id needs to be unique among all config entries of the integration and the entity unique id needs to be unique among all entities for the entity platform of the integration (since the entity registry is aware of both integration domain and platform domain).

Reply:
I've simplified the unique id's based on your feedback.